src/System.Net.HttpListener/tests/Common.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Net.Tests
+{
+    public class GetContextHelper : IDisposable
+    {
+        private HttpClient _client;
+        private HttpListener _listener;
+        private string _listeningUrl;
+
+        public GetContextHelper(HttpListener listener, string listeningUrl)
+        {
+            _listener = listener;
+            _listeningUrl = listeningUrl;
+            _client = new HttpClient();
+        }
+
+        public async Task<HttpListenerResponse> GetResponse()
+        {
+            // We need this to avoid hanging tests.
```

Comment:
Oh, it's in reference to using the client?  That's necessary because GetContextAsync is about handling a request, so if you never get a request, it'll obviously never complete.